primitives/consensus/common/src/import_queue/basic_queue.rs

```diff
@@ -40,9 +41,28 @@ pub struct BasicQueue<B: BlockT, Transaction> {
 	manual_poll: Option<Pin<Box<dyn Future<Output = ()> + Send>>>,
 	/// A thread pool where the background worker is being run.
 	pool: Option<futures::executor::ThreadPool>,
+	pool_guard: Arc<(Mutex<usize>, Condvar)>,
```

Comment:
Any reason not to use a https://docs.rs/crossbeam/0.7.3/crossbeam/sync/struct.WaitGroup.html?

Reply:
1. There's no control over the lifetimes of objects passed to `ThreadPoolBuilder`. It uses some reference counting internally and relying on implementation details seems wrong.
2. Don't want to introduce another dependency over a trivial thing.